app/models/post.rb

```diff
@@ -0,0 +1,19 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+
+  has_many :comments, foreign_key: 'post_id', dependent: :destroy
+  has_many :likes, foreign_key: 'post_id', dependent: :destroy
+
+  after_save :update_user_posts_counter
+
+  def recent_comments
+    Comment.where(post_id: id).order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_user_posts_counter
+    user_posts_count = Post.where(author_id:).count
+    author.update(postsCounter: user_posts_count)
+  end
```

Comment:
:orange_circle: [OPTIONAL] Below is a more concise and clear way to achieve the same result as the previous code: 

  def update_user_posts_counter
    author.increment!(:postsCounter)
  end

Reply:
Thank you for the amazing documentation of comments.
I really appreciate your support.
I would like to discuss this part if you don't mind. As I am using `after_save` to call this method on each instance update. Upon deletion it will be called as well and will leave the counter without updating the right count. on another note, on updating any post, it will increment the post counter by one, which shouldn't be happening.
If I may will not include this in my code.

Reply:
@bienvenuushindi kindly advise.
Thank you in advance

Reply:
@Elerqsousy I appreciate your feedback :+1:

Reply:
@Elerqsousy You are absolutely right :bulb:  :+1: concerning the issue :bug: of the counter not getting updated after a deletion. To address this, it's a good practice to implement a dedicated method to handle the deletion operation, and then you can invoke this method using the after_destroy filter. 

Here's how you can implement it:

```
after_destroy :update_user_posts_counter_on_destroy

def update_user_posts_counter_on_destroy
  author.decrement!(:postsCounter)
end

```

Reply:
@Elerqsousy  :bulb: It's also important to note that, in this case, the preferable filter should be `after_create` rather than `after_save`. `after_save` executes actions after a record is saved, regardless of whether it's a new record (create) or an existing record, whereas `after_create` specifically triggers an action after a new record is successfully created, excluding updates to existing records.

Reply:
In simpler terms:

after_save can run multiple times for the same record if it's updated multiple times.

after_create runs only once, and it's specifically triggered after a new record is successfully created. It doesn't run during updates to existing records.